### ISSUE TITLE
Lack implementation for re-weighting when caculating loss

### ISSUE THREAD

**Issue by 1RZLLL** [HITsz-TMG/KaLM-Embedding#9]
Hi,
First of all, I'd like to express my appreciation for this work.
While exploring the code, I noticed that there doesn't appear to be an implementation for re-weighting the loss in a manner similar to Focal Loss.
Could you kindly clarify if this functionality is implemented somewhere in the code and I may have missed it? If not, would you consider adding such code in the future?
Thank you again for your great work, and I look forward to your feedback.

**Comment by YanshekWoo** [comment]
This is a relatively simple approach to weighting the Cross Entropy Loss. For some implementation references, you can first check the project https://github.com/itakurah/Focal-loss-PyTorch/tree/main.

We aim to integrate this feature into the project code in the next version.
